service/cziti/dns-proxy.go

```diff
@@ -320,7 +317,12 @@ outer:
 		if dnsRetryInterval >= 10000 {
 			dnsRetryInterval = 10000
 		}
-		goto outer
+		goto GetUpstream
+	} else {
+		log.Debugf("Upstream DNS dials succeeded.")
+	}
+	if len(upstreamDnsServers) > 0 && len(dnsUpstreams) != 0 && dnsRetryInterval>500 {
```

Comment:
this whole if can be removed, can't it? also the function called at 325 would have already been called - so this whole block should be removed?